Source/WebCore/animation/ScrollTimeline.cpp

```diff
@@ -121,4 +122,18 @@ Ref<CSSValue> ScrollTimeline::toCSSValue(const RenderStyle&) const
     return CSSScrollValue::create(CSSPrimitiveValue::create(scroller), CSSPrimitiveValue::create(toCSSValueID(m_axis)));
 }
 
+AnimationTimelinesController* ScrollTimeline::controller() const
+{
+    if (m_source)
+        return &m_source->document().ensureTimelinesController();
+    return nullptr;
+}
+
+AnimationTimeline::ShouldUpdateAnimationsAndSendEvents ScrollTimeline::documentWillUpdateAnimationsAndSendEvents()
+{
+    if (m_source)
```

Comment:
I think we should also check that `m_source` is in the tree using `m_source->isConnected()`.

Reply:
Fixed.